compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -636,6 +647,232 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+// ---- pickling untyped trees ----------------------------------
+
+  def pickleUntyped(tree: untpd.Tree)(implicit ctx: Context): Unit = {
+
+    def pickleDummyRef(): Unit = writeNat(0)
+
+    def pickleDummyType(): Unit = {
+      writeByte(SHAREDtype)
+      pickleDummyRef()
+    }
+
+    def pickleUnlessEmpty(tree: untpd.Tree): Unit =
+      if (!tree.isEmpty) pickleUntyped(tree)
+
+    def pickleTpt(tree: untpd.Tree) = pickleUntyped(tree)(ctx.addMode(Mode.Type))
+    def pickleTerm(tree: untpd.Tree) = pickleUntyped(tree)(ctx.retractMode(Mode.Type))
+
+    def pickleAllParams(tree: untpd.DefDef): Unit = {
+      pickleParams(tree.tparams)
+      for (vparams <- tree.vparamss) {
+        writeByte(PARAMS)
+        withLength { pickleParams(vparams) }
+      }
+    }
+
+    def pickleParams(trees: List[untpd.Tree]): Unit =
+      trees.foreach(pickleParam)
+
+    def pickleParam(tree: untpd.Tree): Unit = tree match {
+      case tree: untpd.ValDef => pickleDef(PARAM, tree, tree.tpt)
```

Comment:
Is it intentional to not have the `registerTreeAddr` like in the typed version?

Reply:
Yes. Untyped trees have no cross-references.